### PR TITLE
flavor bugfix

### LINKED ISSUE #176: [BUG] Installer fails, when high availability feature is enabled and flavor large, medium or small is chosen
**Describe the bug**
It seems like the installer is throwing an error when the following constellation is met:
Starting installation with `./openshift-install-powervs create` and set High Availability to yes (choose 1) and choose flavor large, medium or small. When the flavor CUSTOM is chosen, there is no error.

Error msg:
```
[...]
awk: unknown option -i ignored

awk: can't open file { if ( $1 == "bastion" ) $12 = 2 ; print $0 }
 source line number 1
```

When high availability is not enabled (2) then none of the flavors throw an error.

**To Reproduce**
1. Download the openshift-installer-power 
```
(base) mgiessing@Marvins-MBP ocp-install-dir % curl -sL https://raw.githubusercontent.com/ocp-power-automation/openshift-install-power/master/openshift-install-powervs -o openshift-install-powervs  && chmod +x openshift-install-powervs
```

2a. Start installation and enable high availability feature: [This DOES fail]
```
(base) mgiessing@Marvins-MBP ocp-install-dir % ./openshift-install-powervs create
[setup_ibmcloudcli] Installing the latest version of IBM-Cloud CLI...
Attempt: 1/5
/Users/mgiessing/git/ocp-install-dir/ibmcloud version 2.0.0+5a84b9b-2021-06-23T20:06:11+00:00
[setup_terraform] Installing the latest version of Terraform...
Attempt: 1/5
Terraform v0.13.6
[setup_occli] Installing the latest version of OpenShift CLI...
Attempt: 1/5
Client Version: 4.7.18
[precheck_artifacts] WARN: Cannot find artifacts directory... running setup command
[setup] Running setup command...
[setup_artifacts] Downloading code artifacts release-4.7 in ./automation
Attempt: 1/5
[setup] SUCCESS: setup command completed!
[precheck_input] WARN: No variables specified or var.tfvars does not exist.. running variables command
[variables] Trying to login with the provided IBMCLOUD_API_KEY...
[question] > Select the Service Instance name to use: (ocp-fra-test)
? 
- You have answered: ocp-fra-test
Targeting service crn:v1:bluemix:public:power-iaas:eu-de-1:a/d6e204bdd18e25524a9c4c8ee2da34fe:69a4f3ae-c16b-48b6-89f0-e6d2ec858e5c::...
[variables] Gathering information from the selected Service Instance... Please wait
[question] > Select the RHEL image to use for bastion node:
1) Linux-CentOS-8-3
2) Linux-RHEL-SAP-8-1
3) centos-8.3-12102020
#? 3
- You have answered: centos-8.3-12102020
[question] > Select the RHCOS image to use for cluster nodes:
1) rhcos-47-02072021
#? 1
- You have answered: rhcos-47-02072021
[question] > Select the system type to use for cluster nodes:
1) e980
2) s922
#? 2
- You have answered: s922
[question] > Select the private network to use:
1) test-ocp-2a98-eu-de-1
2) ocp-net
#? 2
- You have answered: ocp-net
[question] > Select the OCP version to use:
1) stable-4.7	    8) 4.7.16	      15) 4.7.9		 22) 4.7.2
2) latest-4.7	    9) 4.7.15	      16) 4.7.8		 23) 4.7.1
3) fast-4.7	   10) 4.7.14	      17) 4.7.7		 24) 4.7.0
4) candidate-4.7   11) 4.7.13	      18) 4.7.6		 25) 4.7.0-rc.3
5) 4.7.19	   12) 4.7.12	      19) 4.7.5		 26) 4.7.0-rc.2
6) 4.7.18	   13) 4.7.11	      20) 4.7.4		 27) 4.7.0-rc.1
7) 4.7.17	   14) 4.7.10	      21) 4.7.3
#? 1
- You have answered: stable-4.7
[question] > Enter a short name to identify the cluster (test-ocp)
? 
- You have answered: test-ocp
[question] > Enter a domain name for the cluster (ibm.com)
? 
- You have answered: ibm.com
[question] > Do you want to configure High Availability for bastion nodes?
1) yes
2) no
#? 1
- You have answered: yes
[question] > Select the flavor for the cluster nodes:
1) large
2) medium
3) small
4) CUSTOM
#? 3
- You have answered: small
awk: unknown option -i ignored

awk: can't open file { if ( $1 == "bastion" ) $12 = 2 ; print $0 }
 source line number 1
``` 

In contrast to this see:
2b. Start installation and disable high availability feature: [This does NOT fail]
```
(base) mgiessing@Marvins-MBP ocp-install-dir % ./openshift-install-powervs create
[precheck_input] WARN: No variables specified or var.tfvars does not exist.. running variables command
[variables] Trying to login with the provided IBMCLOUD_API_KEY...
[question] > Select the Service Instance name to use: (ocp-fra-test)
? 
- You have answered: ocp-fra-test
Targeting service crn:v1:bluemix:public:power-iaas:eu-de-1:a/d6e204bdd18e25524a9c4c8ee2da34fe:69a4f3ae-c16b-48b6-89f0-e6d2ec858e5c::...
[variables] Gathering information from the selected Service Instance... Please wait
[question] > Select the RHEL image to use for bastion node:
1) Linux-CentOS-8-3
2) Linux-RHEL-SAP-8-1
3) centos-8.3-12102020
#? 3
- You have answered: centos-8.3-12102020
[question] > Select the RHCOS image to use for cluster nodes:
1) rhcos-47-02072021
#? 1
- You have answered: rhcos-47-02072021
[question] > Select the system type to use for cluster nodes:
1) e980
2) s922
#? 2
- You have answered: s922
[question] > Select the private network to use:
1) test-ocp-2a98-eu-de-1
2) ocp-net
#? 2
- You have answered: ocp-net
[question] > Select the OCP version to use:
1) stable-4.7	    8) 4.7.16	      15) 4.7.9		 22) 4.7.2
2) latest-4.7	    9) 4.7.15	      16) 4.7.8		 23) 4.7.1
3) fast-4.7	   10) 4.7.14	      17) 4.7.7		 24) 4.7.0
4) candidate-4.7   11) 4.7.13	      18) 4.7.6		 25) 4.7.0-rc.3
5) 4.7.19	   12) 4.7.12	      19) 4.7.5		 26) 4.7.0-rc.2
6) 4.7.18	   13) 4.7.11	      20) 4.7.4		 27) 4.7.0-rc.1
7) 4.7.17	   14) 4.7.10	      21) 4.7.3
#? 1
- You have answered: stable-4.7
[question] > Enter a short name to identify the cluster (test-ocp)
? 
- You have answered: test-ocp
[question] > Enter a domain name for the cluster (ibm.com)
? 
- You have answered: ibm.com
[question] > Do you want to configure High Availability for bastion nodes?
1) yes
2) no
#? 2
- You have answered: no
[question] > Select the flavor for the cluster nodes:
1) large
2) medium
3) small
4) CUSTOM
#? 3
- You have answered: small
[question] > Do you need NFS storage to be configured?
1) yes
2) no
#? 
```

**Expected behavior**
I'd expect that everything would work without trowing an error, even when I enable HA

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version BigSur 11.4

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot Parab <Prajyot.Parab1@ibm.com>

Fixes #176

### CODE REVIEW

File: openshift-install-powervs [hunk 1]
```diff
@@ -851,7 +851,13 @@ function variables_nodes {
     question "Select the flavor for the cluster nodes:" "${flavor_list[*]}" yes 
     if [ "${value}" != "CUSTOM" ]; then
       cat automation/compute-vars/$value.tfvars >> "$VAR_TEMPLATE"
-      if [ "${no_of_bastion}" -eq 2 ]; then awk -i inplace '{ if ( $1 == "bastion" ) $12 = 2 ; print $0 }' $VAR_TEMPLATE; fi
+      if [ "${no_of_bastion}" -eq 2 ]; then
+        if [[ "$OS" == "darwin" ]]; then
+          sed -i '' -e 's/\(bastion.*\) 1/\1 2/' $VAR_TEMPLATE;
```
Comment (bpradipt) [inline openshift-install-powervs:856]: Is this to handle non-GNU sed in OSx? 
Does the issue occur if you install gnu sed, make it default in OSX and run the script?

Comment (Prajyot-Parab) [reply openshift-install-powervs:856]: @bpradipt I spoke to @mgiessing and the `gnu-sed` approach isn't working for him, we are good to merge this PR for now.

Comment (bpradipt) [reply openshift-install-powervs:856]: This will fail if gnu sed is used and is default for OSX.
You'll need to check for both darwin and non-GNU sed.

**Review by bpradipt — APPROVED** [review]
/lgtm

**Comment by yussufsh** [comment]
I take a more skeptical view on the provided code, there should be a simpler approach. May be the awk command was at fault?
Such code can lead to more errors in future say when we update the compute files we need to make sure the `bastion.* 1` is always referring to the `count`.

**Review by yussufsh — APPROVED** [review]

**Pull request merged**